/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

syntax = "proto3";

package featureform.embedding.proto;

service EmbeddingHubMeta {
  rpc ListEntries(ListEntriesRequest) returns (stream ListEntriesResponse) {}
}

message ListEntriesRequest {
  //purposely empty
}

message ListEntriesResponse {
  SpaceEntry space = 1;
  repeated VersionEntry version_entry = 2;
}

message SpaceEntry {
  string path = 1;
  string name = 2;
  string default_version = 3;
}

message VersionEntry {
  string path = 1;
  string space = 2;
  string name = 3;
  int32 dims = 4;
  string description = 5;
  string owner = 6;
  repeated string tags = 7;
  string created = 8;
  string revision = 9;
}
